binding_generator.py

```diff
@@ -961,6 +961,8 @@ def generate_engine_class_header(class_api, used_classes, fully_used_classes, us
         result.append("")
         result.append("\ttemplate<class T>")
         result.append("\tstatic T *cast_to(Object *p_object);")
+
+        result.append("\tvirtual ~Object();")
```

Comment:
Why not = default?

Reply:
I originally declared it as `result.append("\tvirtual ~Object() {}") but python didn't like it for some reason, it was late, I wanted to go to bed, so I just took the easiest solution that worked :)

I've never seen the `= default` syntax before though, does that work?

Reply:
Ok, so python complained originally because I was using `result.append(f"\virtual...`, for some reason the curly brackets get parsed at that point. Without that it works so I'll change it :)

Reply:
> I've never seen the = default syntax before though, does that work?

It's a syntactic sugar for constructor / destructor without implementation. Example from Godot codebase:

https://github.com/godotengine/godot/blob/0068247ab68c18653c986358b6949f80c5f10b24/modules/gdscript/gdscript_lambda_callable.h#L62